### PR TITLE
Added more variation to Tiberian Sun debris

### PR DESCRIPTION
**Pull request by Mailaender**
Followup of https://github.com/OpenRA/OpenRA/pull/8686 to add more polish, but less unused trail and sync overhead.

### CODE REVIEW

**Comment by abcdefg30** [comment]
Looks good to me. :+1: / :white_check_mark:

**Comment by abcdefg30** [comment]
Needs a rebase.

File: OpenRA.Mods.Common/Traits/ThrowsShrapnel.cs [hunk 1]
```diff
@@ -19,8 +19,14 @@ public class ThrowsShrapnelInfo : ITraitInfo
 	{
 		[WeaponReference, FieldLoader.Require]
 		public string[] Weapons = { };
+
+		[Desc("Don't fire all weapons at once, but choose one by chance.")]
+		public bool RandomWeapon = false;
```
Comment (Unit158) [inline OpenRA.Mods.Common/Traits/ThrowsShrapnel.cs]: The weapons should have been random originally. It makes no sense to fire more than one weapon as shrapnel at a time, IMO.

File: OpenRA.Mods.Common/Effects/Shrapnel.cs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+namespace OpenRA.Mods.Common.Effects
+{
+	public class ShrapnelInfo : IProjectileInfo
+	{
+		[Desc("Projectile speed in WDist / tick, two values indicate variable velocity.")]
+		public readonly WDist[] Speed = { new WDist(17) };
+
+		[Desc("Maximum offset at the maximum range.")]
+		public readonly WDist Inaccuracy = WDist.Zero;
```
Comment (Unit158) [inline OpenRA.Mods.Common/Effects/Shrapnel.cs]: This should be determined by the ThrowsShrapnel trait. The cohesion here is OK imo, as this is a case-specific optimization.

**Comment by Mailaender** [comment]
Rebased and thinned out the shrapnel weapon even more.

File: mods/d2k/weapons.yaml [hunk 3]
```diff
@@ -699,7 +699,6 @@ Shrapnel:
 		Speed: 50, 125
 		Blockable: false
 		Angle: 91, 264
-		Inaccuracy: 416
```
Comment (reaperrr) [inline mods/d2k/weapons.yaml]: Is there a reason why this still uses `Bullet` instead of `Shrapnel`?

**Comment by reaperrr** [comment]
:+1: once my comment has been adressed.

File: mods/d2k/weapons.yaml (hunk 3 above)
Comment (Mailaender) [inline mods/d2k/weapons.yaml]: Derp. Should be fixed now.

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/0135622feaec14d6ba3a10d12aaecbcd6cad410f...b0c8737009859e1bfae3e236599d77cfab88eb0b)

**Comment by pchote** [comment]
> less unused trail and sync overhead.

Do we have profiling that shows this to be an actual problem?

**Comment by pchote** [comment]
See #9186.

**Pull request merged**